### PR TITLE
23 add project summaries

### PR DESCRIPTION
**Pull request by hmrtn**
fix issue #23 

adds the project_summary boolean flag for the project endpoint. If true - will calculate and return the project summary data. Vote data and project data must exist in the database in order to successfully calculate. 

```shell 
$ curl http://localhost:8080/project\?project_id\=0xc290dd8e51ac35480d9872ce4484aac23bb812c47c0567bfd4beb9113726ed11\&project_summary\=true | jq
```

```shell
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   451  100   451    0     0    291      0  0:00:01  0:00:01 --:--:--   292
{
  "data": null,
  "project_meta_ptr": null,
  "project_votes": null,
  "project_summary": {
    "project_id": "0xc290dd8e51ac35480d9872ce4484aac23bb812c47c0567bfd4beb9113726ed11",
    "round_id": "0xd95a1969c41112cee9a2c931e849bcef36a16f4c",
    "vote_count": 1166,
    "unique_voter_count": 1154,
    "vote_token_sum": [
      {
        "amount": "0x2410c592aceebb8000",
        "token": "0x6b175474e89094c44da98b954eedeac495271d0f"
      },
      {
        "amount": "0x111577b8f0208a9b",
        "token": "0x0000000000000000000000000000000000000000"
      }
    ]
  }
}
```

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -3,3 +3,35 @@ pub mod models;
 pub mod schema;
 pub mod seed;
 pub mod utils;
+
```
Comment (hmrtn) [inline src/lib.rs:6]: This is just an example of what tests should look like. Unit tests and integration tests will likely require some external dev database conn

Comment (vacekj) [reply src/lib.rs:6]: Can't we provision a db in-memory (SQLite?), migrate and seed it in a beforeAll/Each and then tear it down afterwards? Seems better to me than maintaining and pairing for a testing database.

Comment (vacekj) [reply src/lib.rs:6]: Also I don't see any tests in the PR?

Comment (vacekj) [reply src/lib.rs:6]: Like this: https://levelup.gitconnected.com/using-sqlite-with-rust-and-actix-web-with-tests-11a935ac3d95

**Pull request merged**